### PR TITLE
Shareprogress cron

### PR DESCRIPTION
**Pull request by Tuuleh**
I added a couple of specs today and did some clean up. Ready for review now.

### CODE REVIEW

**Comment by Tuuleh** [comment]
About shareprogress analytics:

We're currently using the `share_tests` analytics in the share panel in` pages/:friendly_id:edit`. What I noticed today is that this field in the response gets populated with data only if there is an active A/B test. And active A/B test exists if there are more than one variant of the same share. Consequently, the data about number of shares or conversion rates is not available from this field in the response if there is only one type of the particular share. In other words, if there's only one facebook share, the number of shares is only available from `['response'][0]['share_types']['facebook']`.

File: app/workers/share_analytics_updater.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  private
+
+  def update_shares
+    @collection.each do |button|
+      begin
+        request_analytics_from_sp(button)
+      rescue ::ShareProgressApiError
+        puts $!.message
```
Comment (NealJMD) [inline app/workers/share_analytics_updater.rb]: what's this?

Comment (Tuuleh) [inline app/workers/share_analytics_updater.rb]: This'll print the message in the most recent error.

Comment (NealJMD) [inline app/workers/share_analytics_updater.rb]: wat how

File: cron.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+version: 1
+cron:
+ - name: "update_share_analytics"
+   url: "/update_share_analytics"
+   schedule: "*/5 * * * *"
```
Comment (NealJMD) [inline cron.yaml]: this means every 5 minutes?

File: spec/lib/ak_donation_action_creator_spec.rb [hunk 3]
```diff
@@ -25,8 +25,10 @@
   }
 
   it 'calls the endpoint for creating a donation action in ActionKit' do
-    stub_request(:post, 'https://fake_username:fake_password@act.sumofus.org/rest/v1/action/')
-    subject.create_donation_action(full_donation_options)
-    expect(WebMock).to have_requested(:post, 'https://fake_username:fake_password@act.sumofus.org/rest/v1/donationpush/')
+    VCR.use_cassette('AkDonationActionCreator') do
+      stub_request(:post, 'https://fake_username:fake_password@act.sumofus.org/rest/v1/action/')
+      subject.create_donation_action(full_donation_options)
+      expect(WebMock).to have_requested(:post, 'https://fake_username:fake_password@act.sumofus.org/rest/v1/donationpush/')
```
Comment (NealJMD) [inline spec/lib/ak_donation_action_creator_spec.rb]: isn't it weird to have a WebMock request inside a VCR spec?

File: app/workers/share_analytics_updater.rb (hunk 1 above)
Comment (Tuuleh) [inline app/workers/share_analytics_updater.rb]: When an exception is raised, Ruby places a reference to the Exception object associated with the exception into the global variable`$!`. In this case, this is a `ShareProgressApiError` that has a message associated with it (basically I just print whatever the ShareProgress api sent back in `response.message`).

File: cron.yaml (hunk 2 above)
Comment (Tuuleh) [inline cron.yaml]: Yass

File: spec/lib/ak_donation_action_creator_spec.rb (hunk 3 above)
Comment (Tuuleh) [inline spec/lib/ak_donation_action_creator_spec.rb]: So basically this isn't a spec I wrote - I didn't really have a look at it. I disabled making external requests if a VCR cassette didn't exist, and this spec started failing, so I added a cassette to it as a quick fix. I don't know what the AkDonationActionCreator does and it wasn't something that was touched by my PR, so I didn't fix the spec.

**Comment by NealJMD** [comment]
Looks great!

**Pull request merged**